### PR TITLE
feat: add locator functionality (#238)

### PR DESCRIPTION
**Pull request by masaushi**
This is implementation for `Locator` (#238).
Please review the implementations.

thank you!

### CODE REVIEW

File: locator.go [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+}
+
+func (l *locatorImpl) DispatchEvent(typ string, eventInit interface{}, options ...PageDispatchEventOptions) error {
+	return l.frame.DispatchEvent(l.selector, typ, eventInit, options...)
+}
+
+func (l *locatorImpl) DragTo(target Locator, options ...FrameDragAndDropOptions) error {
+	if len(options) == 1 {
+		strict := true
+		options[0].Strict = &strict
```
Comment (mxschmitt) [inline locator.go:130]:
```suggestion
		options[0].Strict = Bool(true)
```
I think we have a wrapper function for that.

File: locator.go [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+}
+
+func (l *locatorImpl) Evaluate(expression string, arg interface{}, options ...LocatorEvaluateOptions) (interface{}, error) {
+	var option PageWaitForSelectorOptions
+	if len(options) == 1 {
+		option.Timeout = options[0].Timeout
+	}
+
+	var result interface{}
+	err := l.withElement(func(handle ElementHandle) (err error) {
```
Comment (mxschmitt) [inline locator.go:158]: in theory we could change withElement that it returns `(error, interface{})` so we don't need to have the outer variable assignment. Not sure tho if thats better!

File: locator.go [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+}
+
+func (l *locatorImpl) Highlight() error {
+	return l.frame.highlight(l.selector)
+}
+
+func (l *locatorImpl) Hover(options ...PageHoverOptions) error {
+	if len(options) == 1 {
+		strict := true
+		options[0].Strict = &strict
```
Comment (mxschmitt) [inline locator.go:234]: ditto etc. as per above with the Bool(true) comment.

File: tests/locator_test.go [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+	require.NoError(t, page.SetContent(`<div>A</div><div>B</div><div>C</div>`))
+
+	locator, err := page.Locator("div")
+	require.NoError(t, err)
+	innerHTML, err := locator.AllInnerTexts()
+	require.NoError(t, err)
+	require.ElementsMatch(t, []string{"A", "B", "C"}, innerHTML)
+}
+
+func TestLocatorAllTextContens(t *testing.T) {
```
Comment (mxschmitt) [inline tests/locator_test.go:24]:
```suggestion
func TestLocatorAllTextContents(t *testing.T) {
```

**Review by mxschmitt — APPROVED** [review]
@masaushi you did an amazing job there, I really appreciate your time and effort which you put into this! 💯 Thank you for contributing to the project, all the comments are just small nits.

File: locator.go (hunk 1 above)
Comment (masaushi) [reply locator.go]: I didn't know that useful wrapper function!
fixed codes including similar to this!

**Comment by mxschmitt** [comment]
Amazing, bots are green and that means your patch related changes were good!

File: locator.go [hunk 5]
```diff
@@ -0,0 +1,412 @@
...
+	}
+
+	return l.withElement(func(handle ElementHandle) (interface{}, error) {
+		return handle.EvaluateHandle(expression, arg)
+	}, option)
+}
+
+func (l *locatorImpl) Fill(value string, options ...FrameFillOptions) error {
+	if len(options) == 1 {
+		options[0].Strict = Bool(true)
```
Comment (evanlurvey) [inline locator.go:166]: This forcefully overwrites whatever config that is passed in unless they pass 2 configs in. Is that intentional?

**Pull request merged**